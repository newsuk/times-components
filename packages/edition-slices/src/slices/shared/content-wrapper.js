/**
 * ContentWrapper restricts the content area of each slice. It is used together with the Gutter component,
 * which sets the gutter space around the content. This component is currently used only for huge breakpoints.
 */

import React from "react";
import PropTypes from "prop-types";
import { TcView } from "@times-components/utils";
import styleFactory from "./styles";

const styles = styleFactory();

const ContentWrapper = ({ children }) => (
  <TcView style={styles.contentWrapperStyles}>{children}</TcView>
);

ContentWrapper.propTypes = {
  children: PropTypes.node.isRequired
};

export default ContentWrapper;
