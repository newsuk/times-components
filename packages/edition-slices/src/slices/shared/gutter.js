import React from "react";
import PropTypes from "prop-types";
import { View } from "react-native";
import { editionMaxWidth } from "@times-components/styleguide";

const Gutter = ({ children }) => (
  <View
    style={{
      alignSelf: "center",
      maxWidth: "100%",
      width: editionMaxWidth
    }}
  >
    {children}
  </View>
);

Gutter.propTypes = {
  children: PropTypes.node.isRequired
};

export default Gutter;
