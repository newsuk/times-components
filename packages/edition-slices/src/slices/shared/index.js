import Gutter from "./gutter";
import ContentWrapper from "./content-wrapper";
import ResponsiveSlice from "./responsive-slice";

export { Gutter, ContentWrapper, ResponsiveSlice };
