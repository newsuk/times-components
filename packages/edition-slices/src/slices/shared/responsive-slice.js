import React from "react";
import PropTypes from "prop-types";
import { ResponsiveContext } from "@times-components/responsive";
import { editionBreakpoints } from "@times-components/ts-styleguide";
import Gutter from "./gutter";
import ContentWrapper from "./content-wrapper";

const ResponsiveSlice = ({
  renderSmall,
  renderMedium,
  renderWide,
  renderHuge
}) => (
  <ResponsiveContext.Consumer>
    {({ editionBreakpoint }) => {
      switch (editionBreakpoint) {
        case editionBreakpoints.small:
          return renderSmall(editionBreakpoint);
        case editionBreakpoints.medium:
          return <Gutter>{renderMedium(editionBreakpoint)}</Gutter>;
        case editionBreakpoints.wide:
          return (
            <Gutter>
              {(renderWide && renderWide(editionBreakpoint)) ||
                (renderMedium && renderMedium(editionBreakpoint))}
            </Gutter>
          );
        case editionBreakpoints.huge:
          return (
            <Gutter>
              <ContentWrapper>
                {(renderHuge && renderHuge(editionBreakpoint)) ||
                  (renderWide && renderWide(editionBreakpoint)) ||
                  (renderMedium && renderMedium(editionBreakpoint))}
              </ContentWrapper>
            </Gutter>
          );
        default:
          return renderSmall(editionBreakpoint);
      }
    }}
  </ResponsiveContext.Consumer>
);

ResponsiveSlice.propTypes = {
  renderSmall: PropTypes.func.isRequired,
  renderMedium: PropTypes.func.isRequired,
  renderWide: PropTypes.func,
  renderHuge: PropTypes.func
};

ResponsiveSlice.defaultProps = {
  renderWide: null,
  renderHuge: null
};

export default ResponsiveSlice;
