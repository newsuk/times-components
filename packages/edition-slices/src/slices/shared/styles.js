import {
  editionMaxWidth,
  sliceContentMaxWidth,
  spacing,
  editionBreakpoints
} from "@times-components/ts-styleguide";

const tabletPaddingHorizontalResolver = {
  [editionBreakpoints.medium]: spacing(2),
  [editionBreakpoints.wide]: spacing(6),
  [editionBreakpoints.huge]: spacing(2)
};

export default (isTablet, breakpoint) => ({
  contentWrapperStyles: {
    flex: 1,
    alignSelf: "center",
    width: sliceContentMaxWidth
  },

  gutterStyles: {
    alignSelf: "center",
    maxWidth: "100%",
    paddingHorizontal: isTablet
      ? tabletPaddingHorizontalResolver[breakpoint]
      : 0,
    width: editionMaxWidth
  }
});
