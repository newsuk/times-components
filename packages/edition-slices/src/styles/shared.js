const sharedStyle = {
  EditionSlicesBody: {
    padding: 20
  }
};

export default sharedStyle;
