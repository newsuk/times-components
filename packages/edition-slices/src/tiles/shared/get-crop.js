export default (leadAsset, crop) => {
  if (!leadAsset) return null;

  // eslint-disable-next-line no-underscore-dangle
  const isVideo = leadAsset.__typename === "Video";
  const displayImage = isVideo
    ? leadAsset.posterImage[crop] || {}
    : leadAsset[crop] || {};

  return displayImage.url;
};
