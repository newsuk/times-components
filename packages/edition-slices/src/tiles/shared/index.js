import getCrop from "./get-crop";
import TileImage from "./tile-image";
import TileLink from "./tile-link";
import TileSummary from "./tile-summary";
import withTileTracking from "./tile-tracking-events";
import withPuzzleTileTracking from "./puzzle-tracking-events";

export {
  getCrop,
  TileImage,
  TileLink,
  TileSummary,
  withPuzzleTileTracking,
  withTileTracking
};
