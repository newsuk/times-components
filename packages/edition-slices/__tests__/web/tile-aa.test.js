import shared from "../tile-aa/shared-tile-aa.web";

shared();
