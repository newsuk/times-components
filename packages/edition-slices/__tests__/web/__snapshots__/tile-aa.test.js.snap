// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tile aа huge 1`] = `
<style>
.S1 {
  margin-bottom: 0px;
}

.S2 {
  color: rgba(51,51,51,1.00);
  font-family: TimesModern-Bold;
  font-weight: 400;
}

.IS1 {
  font-size: 28px;
  line-height: 28px;
  margin-bottom: 10px;
}

.IS2 {
  -webkit-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  flex-basis: 0%;
  -ms-flex-positive: 1;
  -webkit-box-flex: 1;
  -ms-flex-negative: 1;
  -ms-flex-preferred-size: 0%;
}
</style>

<Link
  linkStyle={
    Object {
      "flex": 1,
      "paddingHorizontal": "10px",
      "paddingVertical": "15px",
    }
  }
  url="/article/123"
>
  <div
    className="css-view-1dbjc4n IS2"
  >
    <div
      className="css-view-1dbjc4n"
    >
      <div
        className="css-view-1dbjc4n"
      >
        <div
          className="css-view-1dbjc4n r-marginBottom-p1pxzi S1"
        >
          <ArticleLabel
            color="#005B8D"
            isVideo={false}
            title="label"
          />
        </div>
        <h3
          aria-level="3"
          className="css-reset-4rbku5 css-text-901oao r-color-1khnkhu r-fontFamily-iirzy8 r-fontWeight-16dba41 IS1 S2"
          role="heading"
        >
          This is tile headline
        </h3>
        <ArticleFlags
          flags={
            Array [
              Object {
                "expiryTime": "2030-03-14T12:00:00.000Z",
                "type": "EXCLUSIVE",
              },
            ]
          }
        />
      </div>
    </div>
  </div>
</Link>
`;

exports[`tile aа medium 1`] = `
<style>
.S1 {
  margin-bottom: 0px;
}

.S2 {
  color: rgba(51,51,51,1.00);
  font-family: TimesModern-Bold;
  font-weight: 400;
  margin-bottom: 5px;
}

.IS1 {
  font-size: 20px;
  line-height: 20px;
}

.IS2 {
  -webkit-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  flex-basis: 0%;
  -ms-flex-positive: 1;
  -webkit-box-flex: 1;
  -ms-flex-negative: 1;
  -ms-flex-preferred-size: 0%;
}
</style>

<Link
  linkStyle={
    Object {
      "flex": 1,
      "paddingHorizontal": "10px",
      "paddingVertical": "15px",
    }
  }
  url="/article/123"
>
  <div
    className="css-view-1dbjc4n IS2"
  >
    <div
      className="css-view-1dbjc4n"
    >
      <div
        className="css-view-1dbjc4n"
      >
        <div
          className="css-view-1dbjc4n r-marginBottom-p1pxzi S1"
        >
          <ArticleLabel
            color="#005B8D"
            isVideo={false}
            title="label"
          />
        </div>
        <h3
          aria-level="3"
          className="css-reset-4rbku5 css-text-901oao r-color-1khnkhu r-fontFamily-iirzy8 r-fontWeight-16dba41 r-marginBottom-d0pm55 IS1 S2"
          role="heading"
        >
          This is tile headline
        </h3>
        <ArticleFlags
          flags={
            Array [
              Object {
                "expiryTime": "2030-03-14T12:00:00.000Z",
                "type": "EXCLUSIVE",
              },
            ]
          }
        />
      </div>
    </div>
  </div>
</Link>
`;

exports[`tile aа wide 1`] = `
<style>
.S1 {
  margin-bottom: 0px;
}

.S2 {
  color: rgba(51,51,51,1.00);
  font-family: TimesModern-Bold;
  font-weight: 400;
}

.IS1 {
  font-size: 28px;
  line-height: 28px;
  margin-bottom: 10px;
}

.IS2 {
  -webkit-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  flex-basis: 0%;
  -ms-flex-positive: 1;
  -webkit-box-flex: 1;
  -ms-flex-negative: 1;
  -ms-flex-preferred-size: 0%;
}
</style>

<Link
  linkStyle={
    Object {
      "flex": 1,
      "paddingHorizontal": "10px",
      "paddingVertical": "15px",
    }
  }
  url="/article/123"
>
  <div
    className="css-view-1dbjc4n IS2"
  >
    <div
      className="css-view-1dbjc4n"
    >
      <div
        className="css-view-1dbjc4n"
      >
        <div
          className="css-view-1dbjc4n r-marginBottom-p1pxzi S1"
        >
          <ArticleLabel
            color="#005B8D"
            isVideo={false}
            title="label"
          />
        </div>
        <h3
          aria-level="3"
          className="css-reset-4rbku5 css-text-901oao r-color-1khnkhu r-fontFamily-iirzy8 r-fontWeight-16dba41 IS1 S2"
          role="heading"
        >
          This is tile headline
        </h3>
        <ArticleFlags
          flags={
            Array [
              Object {
                "expiryTime": "2030-03-14T12:00:00.000Z",
                "type": "EXCLUSIVE",
              },
            ]
          }
        />
      </div>
    </div>
  </div>
</Link>
`;

exports[`tile aа without breakpoint should be like medium 1`] = `
<style>
.S1 {
  margin-bottom: 0px;
}

.S2 {
  color: rgba(51,51,51,1.00);
  font-family: TimesModern-Bold;
  font-weight: 400;
  margin-bottom: 5px;
}

.IS1 {
  font-size: 20px;
  line-height: 20px;
}

.IS2 {
  -webkit-flex-grow: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  flex-shrink: 1;
  -webkit-flex-basis: 0%;
  flex-basis: 0%;
  -ms-flex-positive: 1;
  -webkit-box-flex: 1;
  -ms-flex-negative: 1;
  -ms-flex-preferred-size: 0%;
}
</style>

<Link
  linkStyle={
    Object {
      "flex": 1,
      "paddingHorizontal": "10px",
      "paddingVertical": "15px",
    }
  }
  url="/article/123"
>
  <div
    className="css-view-1dbjc4n IS2"
  >
    <div
      className="css-view-1dbjc4n"
    >
      <div
        className="css-view-1dbjc4n"
      >
        <div
          className="css-view-1dbjc4n r-marginBottom-p1pxzi S1"
        >
          <ArticleLabel
            color="#005B8D"
            isVideo={false}
            title="label"
          />
        </div>
        <h3
          aria-level="3"
          className="css-reset-4rbku5 css-text-901oao r-color-1khnkhu r-fontFamily-iirzy8 r-fontWeight-16dba41 r-marginBottom-d0pm55 IS1 S2"
          role="heading"
        >
          This is tile headline
        </h3>
        <ArticleFlags
          flags={
            Array [
              Object {
                "expiryTime": "2030-03-14T12:00:00.000Z",
                "type": "EXCLUSIVE",
              },
            ]
          }
        />
      </div>
    </div>
  </div>
</Link>
`;
